### ISSUE TITLE
batch_encode_plus with pad_to_max_length is not padding the output

### ISSUE THREAD

**Issue by dipanjannag** [huggingface/transformers#3012]
# 🐛 Bug

## Information

Model I am using (BertTokenizer,):

Language I am using the model on: English

The problem arises when using:
* [ ] the official example scripts.

The tasks I am working on is:
* [ ] Simple batch tokenization task

## To reproduce
```
from transformers import BertTokenizer, BertForQuestionAnswering
import torch

import numpy as np

tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
model = BertForQuestionAnswering.from_pretrained('bert-large-uncased-whole-word-masking-finetuned-squad')

qt1 = ('What is the name of the repository?', 'pipeline have been included in the huggingface/transformers repository')
qt2 = ('What is the name of the repository?', 'What can be done with this???')

inp_raw = [qt1, qt2]
tck_temp = tokenizer.batch_encode_plus(inp_raw, max_length=20, pad_to_max_length=True)

print(tck_temp)
inp_ids = tck_temp['input_ids']
tck_type_ids = tck_temp['token_type_ids']


print(len(inp_ids[0]) == len(inp_ids[1]))
## This is coming false
```

## Expected behavior

The code snippet should print True
But it prints False

## Environment info
     
- `transformers` version: 2.3
- Platform: Linux
- Python version: 3.6
- PyTorch version (1.4CPU)

[EDIT] : correct transformers version

**Comment by TevenLeScao** [comment]
The code snippet prints `True` for me.
I have the versions:
transformers version: 2.5.1
Platform: Linux
Python version: 3.6.9
PyTorch version: 1.4.0+cpu
Can you try `pip install --upgrade transformers` and see whether the error is still there?

**Comment by dipanjannag** [comment]
Thanks @TevenLeScao it seems the issue was only for transformers ~2.3
